packages/client/src/clients/guide/client.ts

```diff
@@ -433,7 +459,10 @@ export class KnockGuideClient {
 
     // If a guide ignores the group limit, then return immediately to render
     // always.
-    if (guide.bypass_global_group_limit) {
+    if (
+      guide.bypass_global_group_limit ||
+      state.debug.forcedGuideKey === guide.key
+    ) {
       return guide;
```

Comment:
I think you may need to also intercept the order resolution logic when we resolve the prevailing guide in closePendingGroupStage()? Meaning, "regardless of the order configured, force this guide to render ahead of any guides".